src/core/config/mac.h

```diff
@@ -139,6 +139,16 @@
 #define OPENTHREAD_CONFIG_MAC_MAX_TX_ATTEMPTS_INDIRECT_POLLS 4
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_MAC_CSL_REQUEST_AHEAD_THRESHOLD_US
+ *
+ * Define how many microseconds ahead should MAC deliver CSL frame to SubMac.
```

Comment:
Where does this number come from? How can the user calculate it for their platform?

Reply:
This is not a platform-specific value. It's used to minimize the exception cases that MAC is somehow delayed to pass CSL frame to SubMac, and SubMac has to send the frame in next slot. So theoretically it cannot 100% ensure this. It's an experienced value.